python/ml_croissant/ml_croissant/_src/structure_graph/nodes/source.py

```diff
@@ -106,58 +128,81 @@ class Source:
     extract: Extract = dataclasses.field(default_factory=Extract)
     transforms: tuple[Transform, ...] = ()
     uid: str | None = None
+    node_type: Literal["distribution", "field"] | None = None
+
+    def to_json(self) -> Json:
+        """Converts the `Source` to JSON."""
+        transforms = [transform.to_json() for transform in self.transforms]
+        if len(transforms) == 1:
+            transforms = transforms[0]
+        return remove_empty_values(
+            {
+                self.node_type: self.uid,
+                "applyTransform": transforms,
+                "dataExtraction": self.extract.to_json(),
+            }
+        )
 
     @classmethod
-    def from_json_ld(cls, issues: Issues, json_ld: Any) -> Source:
+    def from_jsonld(cls, issues: Issues, jsonld: Any) -> Source:
         """Creates a new source from a JSON-LD `field` and populates issues."""
-        if isinstance(json_ld, list):
-            if len(json_ld) != 1:
-                raise ValueError(f"Field {json_ld} should have one element.")
-            return Source.from_json_ld(issues, json_ld[0])
-        elif isinstance(json_ld, (dict, collections.defaultdict)):
+        if jsonld is None:
+            return Source()
+        elif isinstance(jsonld, list):
+            if len(jsonld) != 1:
+                raise ValueError(f"Field {jsonld} should have one element.")
+            return Source.from_jsonld(issues, jsonld[0])
+        elif isinstance(jsonld, dict):
             try:
-                transforms = json_ld.get(constants.CROISSANT_APPLY_TRANSFORM, [])
+                transforms = jsonld.get(str(constants.ML_COMMONS_APPLY_TRANSFORM), [])
                 if not isinstance(transforms, list):
-                    raise ValueError(
-                        'Field "apply_transform" should be parsed as a list.'
-                    )
+                    transforms = [transforms]
```

Comment:
Shouldn't we check the type of `transforms`?

Reply:
Done.

See new commit 
`Better parsing for Transform` (https://github.com/mlcommons/croissant/pull/169/commits/207ba28fd4165bed04a178b8b6f0d53385a093aa).